File: typescript/examples/get-leaderboards.ts
import { MythicMC } from '../src/index.ts'

const api = new MythicMC({ apiKey: process.env.MYTHICMC_API_KEY!, baseUrl: process.env.MYTHICMC_API_URL })

const index = await api.listLeaderboards()
for (const [type, periods] of Object.entries(index.boards)) console.log(`${type}: ${periods.join(', ')}`)

const board = await api.getLeaderboard('kills', 'weekly')
// Periods use America/New_York time; weeks start on Friday.
console.log(`\nkills / weekly, ${board.windowStart} to ${board.windowEnd}`)
if (board.stale) console.log('this board is stale')

// Values are formatted strings. Use player stats for calculations.
for (const row of board.rows) console.log(`#${row.rank} ${row.name}: ${row.value}`)

File: python/examples/get_leaderboards.py
import os

from mythicmc import DEFAULT_BASE_URL, MythicMC

key = os.environ["MYTHICMC_API_KEY"]
base_url = os.environ.get("MYTHICMC_API_URL", DEFAULT_BASE_URL)

with MythicMC(key, base_url=base_url) as api:
    index = api.list_leaderboards()
    board = api.get_leaderboard("kills", "weekly")

for board_type, periods in index.boards.items():
    print(f"{board_type}: {', '.join(periods)}")

# Periods use America/New_York time; weeks start on Friday.
print(f"\nkills / weekly, {board.window_start} to {board.window_end}")
if board.stale:
    print("this board is stale")

# Values are formatted strings. Use player stats for calculations.
for row in board.rows:
    print(f"#{row.rank} {row.name}: {row.value}")
